packages/deployment/scripts/capture-integration-results.sh

```diff
@@ -16,38 +16,51 @@ for node in validator{0,1}; do
   "$thisdir/setup.sh" ssh "$node" cat "$home/data/swingstore-trace" > "$RESULTSDIR/$node-swingstore-trace" || true
 done
 
-val0len=$(wc -l < $RESULTSDIR/validator0-swingstore-trace)
-val1len=$(wc -l < $RESULTSDIR/validator1-swingstore-trace)
-commonlen=$((val0len < val1len ? val0len : val1len))
-
 ret=0
-diff -U0 <(head -n $commonlen $RESULTSDIR/validator0-swingstore-trace) <(head -n $commonlen $RESULTSDIR/validator1-swingstore-trace) > $RESULTSDIR/validator-swingstore-trace.diff || ret=$?
+"$thisdir/../../../scripts/process-integration-swingstore-traces.sh" "$RESULTSDIR" || ret=$?
 
 failedtest=${1:-"unknown"}
 
-if [ $ret -eq 0 ]
-then
-  if [ "$failedtest" = "false" ]
+if [ -f "$RESULTSDIR/divergent_snapshots" ]; then
+  if [ $(cat "$RESULTSDIR/validator-swingstore-trace.diff" | wc -l) -gt 0 ]; then
```

Comment:
Avoid catting the whole file if you're just testing for nonzero size:

```suggestion
  if [ -s "$RESULTSDIR/validator-swingstore-trace.diff" ]; then
```

Reply:
I did not know about this one.